tableauserverclient/models/schedule_item.py

```diff
@@ -0,0 +1,203 @@
+        self.interval_item = interval_item
+
+        # Invoke setter
+        self.execution_order = execution_order
+        self.name = name
+        self.priority = priority
+        self.schedule_type = schedule_type
+
+    @property
+    def created_at(self):
+        return self._created_at
+
+    @property
+    def end_schedule_at(self):
+        return self._end_schedule_at
+
+    @property
+    def execution_order(self):
+        return self._execution_order
+
+    @execution_order.setter
+    def execution_order(self, value):
+        if value and not hasattr(ScheduleItem.ExecutionOrder, value):
+            error = "Invalid execution order defined: {}.".format(value)
+            raise ValueError(error)
+        else:
+            self._execution_order = value
+
+    @property
+    def frequency(self):
+        return self._frequency
+
+    @property
+    def id(self):
+        return self._id
+
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        if not value:
+            error = "Name must be defined."
+            raise ValueError(error)
+        else:
+            self._name = value
+
+    @property
+    def next_run_at(self):
+        return self._next_run_at
+
+    @property
+    def priority(self):
+        return self._priority
+
+    @priority.setter
+    def priority(self, value):
+        if value < 1 or value > 100:
+            error = "Invalid priority defined: {}.".format(value)
+            raise ValueError(error)
+        else:
+            self._priority = value
+
+    @property
+    def schedule_type(self):
+        return self._schedule_type
+
+    @schedule_type.setter
+    def schedule_type(self, value):
+        if not value:
+            error = "Schedule type must be defined."
+            raise ValueError(error)
+        elif not hasattr(ScheduleItem.Type, value):
+            error = "Invalid schedule type defined: {}.".format(value)
+            raise ValueError(error)
+        else:
+            self._schedule_type = value
+
+    @property
+    def state(self):
+        return self._state
+
+    @state.setter
+    def state(self, value):
+        if not hasattr(ScheduleItem.State, value):
+            error = "Invalid state defined."
+            raise ValueError(error)
+        else:
+            self._state = value
+
+    @property
+    def updated_at(self):
+        return self._updated_at
+
+    def _parse_common_tags(self, schedule_xml):
+        if not isinstance(schedule_xml, ET.Element):
+            schedule_xml = ET.fromstring(schedule_xml).find('.//t:schedule', namespaces=NAMESPACE)
+        if schedule_xml is not None:
+            (_, name, _, _, updated_at, _, frequency, next_run_at, end_schedule_at, execution_order,
+             priority, interval_item) = self._parse_element(schedule_xml)
+
+            self._set_values(None, name, None, None, updated_at, None, frequency, next_run_at, end_schedule_at,
```

Comment:
Can you use keyword arguments for all of these -- else we're doomed to be lost at sea :)

(Side note, can any of these be moved out of the function, it seems crazy big)